src/transformState.js

```diff
@@ -5,7 +5,23 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const key in actions) {
+    if (actions[key].type === 'addProperties') {
+      const extraData = Object.assign({}, actions[key].extraData);
+
+      Object.assign(state, extraData);
+    } else if (actions[key].type === 'removeProperties') {
+      const keysToRemove = actions[key].keysToRemove;
+
+      for (const value in keysToRemove) {
+        delete state[keysToRemove[value]];
+      }
+    } else if (actions[key].type === 'clear') {
+      for (const key1 in state) {
+        delete state[key1];
+      }
```

Comment:
Avoid such nesting else if conditions
This code is hard to read and maintain.

In each block "if" and "else if" you compare one value, there is better suited here switch/case